### PR TITLE
[DOCS-347] Update pentester articles round 2.5

### PR DESCRIPTION
**Pull request by mjang-cobalt**
## Changelog

Including links to pentester articles for Harsh and Farid
- Also including content from https://github.com/cobalthq/cobalt-product-public-docs/pull/204/files, as I don't have privileges to modify the file name on Farid's fork

### Added

n/a

### Updated

| Page | Deploy Preview | Comment |
| ----- | ----- | ----- |
| Best Practices page | [link](https://deploy-preview-247--cobalt-docs.netlify.app/bestpractices/) | Comment |

## Preview This Change

To see how this change looks in production, scroll down to **Deploy Preview**. Select the link that looks like `https://deploy-preview-<num>--cobalt-docs.netlify.app/`

## Variables

Help us support a “Write once, publish everywhere” single source of truth. If you see a line that looks like:

`{{% asset-categories %}}`

You’ve found a [shortcode](https://gohugo.io/content-management/shortcodes/) that we include in multiple documents.

You’ll find the content of the shortcode in the following directory:

https://github.com/cobalthq/cobalt-product-public-docs/tree/main/layouts/shortcodes

That shortcode has the same base name as what you see in the PR, such as `asset-categories.html`.

## Checklist for PR Author

[ ] Did you check for broken links and alt text?

Be sure to check for broken links and Alt Text issues. We have a partially automated process,
as described in this section of our repository README:
[Test Links and Alt Attributes](https://github.com/cobalthq/cobalt-product-public-docs/blob/main/README.md#test-links-and-alt-attributes).

### CODE REVIEW

File: content/en/BestPractices/_index.md [hunk 1]
```diff
@@ -25,6 +25,13 @@ until we have enough articles to actually organize. -->
 Once we've merged content into this guide, we'll list them in the following table with the
 date of release.
 
+### December 1, 2022
+
+| Article                                                               | Release Date | Author                |
+|-----------------------------------------------------------------------|--------------|-----------------------|
+| [Insecure Deserialization](./insecure-deserialization)                | 2022-12-01   | {{% farid %}}         |
+| [Protect Against XXE Attacks](./protect-against-xxe)                  | 2022-12-01   | {{% harsh-bothra %}}  |
```
Comment (mjang-cobalt) [inline content/en/BestPractices/_index.md:33]:
```suggestion
| Article | Release Date | Author |
| ----- | ----- | ----- |
|[Insecure Deserialization](./insecure-deserialization)                | 2022-12-01   | {{% farid %}} |
|[Protect Against XXE Attacks](./protect-against-xxe)                | 2022-12-01   | {{% farid %}} |
```

File: content/en/BestPractices/_index.md [hunk 2]
```diff
@@ -25,6 +25,13 @@ until we have enough articles to actually organize. -->
 Once we've merged content into this guide, we'll list them in the following table with the
 date of release.
 
+### December 1, 2022
+
+| Article | Release Date | Author |
+| ----- | ----- | ----- |
+|[Insecure Deserialization](./insecure-deserialization)                | 2022-12-01   | {{% farid %}} |
+|[Protect Against XXE Attacks](./protect-against-xxe)                | 2022-12-01   | {{% harsh-bothra %}} |
```
Comment (mjang-cobalt) [inline content/en/BestPractices/_index.md:33]:
```suggestion
|[Protect Against XXE Attacks](./protect-against-xxe)                | 2022-12-01   | {{% harsh-bothra %}} |
|[Insecure Deserialization](./insecure-deserialization)                | 2022-12-01   | {{% farid %}} |
```

**Comment by mjang-cobalt** [comment]
Grr, we still have problems with formatting Markdown tables, and I've already spent an hour trying different solutions. We need to accept this issue, and investigate. I've filed an internal issue for that purpose.

**Comment by ana-dashuk-cobalt** [comment]
> Grr, we still have problems with formatting Markdown tables, and I've already spent an hour trying different solutions. We need to accept this issue, and investigate. I've filed an internal issue for that purpose.

@mjang-cobalt This is solved. The problem was an extra space in Markdown files with shortcodes.

**Comment by mjang-cobalt** [comment]
> @mjang-cobalt This is solved. The problem was an extra space in Markdown files with shortcodes.

Brilliant, thank you!

File: content/en/BestPractices/insecure-deserialization.md [hunk 3]
```diff
@@ -0,0 +1,290 @@
+---
+Title: "Insecure Deserialization"
+linkTitle: "Insecure Deserialization"
+weight: 660
+description: >
+  Insecure Deserialization is one of the OWASP Top 10 web vulnerabilities for 2017. OWASP has included this issue in the [OWASP Top 10:2021](https://owasp.org/Top10/) vulnerabilities as [Software and Data Integrity Failures](https://owasp.org/Top10/A08_2021-Software_and_Data_Integrity_Failures/).
```
Comment (ana-dashuk-cobalt) [inline content/en/BestPractices/insecure-deserialization.md:6]:
```suggestion
  Insecure Deserialization is one of the OWASP Top 10 web vulnerabilities for 2017. OWASP has included this issue in the [OWASP Top 10:2021](https://owasp.org/Top10/) vulnerabilities as [Software and Data Integrity Failures](https://owasp.org/Top10/A08_2021-Software_and_Data_Integrity_Failures/).
```

This description is too long. We usually use a short sentence.

File: content/en/BestPractices/insecure-deserialization.md [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+
+{{% /pageinfo %}}
+
+Before we can describe security issues in this article, we should define serialization and deserialization.
+
+## What is Serialization?
+
+In serialization, we convert an object to a byte stream. We can then send that data over a network or save it on a disk. Serialization is common with Machine Learning data. We can send pre-trained models to other systems. One common example is the pickle.dumps() function in Python3.
+
+Sample code:
```
Comment (ana-dashuk-cobalt) [inline content/en/BestPractices/insecure-deserialization.md:21]:
```suggestion
Sample code:

```

File: content/en/BestPractices/insecure-deserialization.md [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+
+```
+b'\x80\x04\x950\x00\x00\x00\x00\x00\x00\x00}\x94(\x8c\x04Name\x94\x8c\x08John Doe\x94\x8c\x03Age\x94K\x19\x8c\x05Score\x94G@T\x1333333u.'
+```
+
+## What is Deserialization?
+
+Deserialization is the opposite of Serialization. It is the process of converting a byte stream into a program object. One common example is the `pickle.loads() function in Python3.
+
+Sample Code:
```
Comment (ana-dashuk-cobalt) [inline content/en/BestPractices/insecure-deserialization.md:39]:
```suggestion
Sample code: 

```

File: content/en/BestPractices/insecure-deserialization.md [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+
+Output:
+
+```
+b'\x80\x04\x950\x00\x00\x00\x00\x00\x00\x00}\x94(\x8c\x04Name\x94\x8c\x08John Doe\x94\x8c\x03Age\x94K\x19\x8c\x05Score\x94G@T\x1333333u.'
+```
+
+## What is Deserialization?
+
+Deserialization is the opposite of Serialization. It is the process of converting a byte stream into a program object. One common example is the `pickle.loads() function in Python3.
```
Comment (ana-dashuk-cobalt) [inline content/en/BestPractices/insecure-deserialization.md:37]:
```suggestion
Deserialization is the opposite of Serialization. It is the process of converting a byte stream into a program object. One common example is the `pickle.loads()` function in Python3.
```

File: content/en/BestPractices/insecure-deserialization.md [hunk 7]
```diff
@@ -0,0 +1,290 @@
...
+Sample Code: 
+```python
+import pickle
+
+serialized_data = b'\x80\x04\x950\x00\x00\x00\x00\x00\x00\x00}\x94(\x8c\x04Name\x94\x8c\x08John Doe\x94\x8c\x03Age\x94K\x19\x8c\x05Score\x94G@T\x1333333u.'
+my_dict = pickle.loads(serialized_data)
+print(my_dict)
+```
+
+Output:
```
Comment (ana-dashuk-cobalt) [inline content/en/BestPractices/insecure-deserialization.md:48]:
```suggestion
Output:

```

**Comment by ana-dashuk-cobalt** [comment]
@mjang-cobalt Has the _Insecure Deserialization_ article gone through your review? I see that there are many formatting and language issues, and the readability score is 11.22.

File: content/en/BestPractices/insecure-deserialization.md (hunk 3 above)
Comment (mjang-cobalt) [reply content/en/BestPractices/insecure-deserialization.md:6]: It's two separate sentences.

Comment (ana-dashuk-cobalt) [reply content/en/BestPractices/insecure-deserialization.md:6]: @mjang-cobalt Yes, but it looks terrible in previews and doesn't align with other articles.

Screenshot 2022-12-06 at 20 45 10

**Pull request merged**